### PR TITLE
Support entrypoints defined in a namespace.

### LINKED ISSUE #2144: Namespace Functions not supported as EntryPoint
```
struct PSInput
{
	float4 color : COLOR;
};

namespace Pixel
{
	float4 Main(PSInput input) : SV_TARGET
	{
		return input.color;
	}    
}
```
In the following example, specifying `Pixel::Main` as the entry point fails to compile in Slang, but will work in HLSL compilers FXC and DXC.

### PR DESCRIPTION
**Pull request by csyonghe**
Closes #2144.

### CODE REVIEW

File: tests/language-feature/namespaces/entrypoint-in-namespace.slang [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+namespace Pixel
+{
+    struct MyType
+    {
+        static float4 Main(PSInput input) : SV_TARGET
+        {
+            return input.color;
+        }
+    }    
+}
```
Comment (ArielG-NV) [inline tests/language-feature/namespaces/entrypoint-in-namespace.slang:20]: From what I understand this PR also allows us to define entry-points inside a `struct` (but not inside a namespace), should we add that to this test?

**Review by ArielG-NV — APPROVED** [review]
Code looks good to me, 1 question.

File: tests/language-feature/namespaces/entrypoint-in-namespace.slang (hunk 1 above)
Comment (csyonghe) [reply tests/language-feature/namespaces/entrypoint-in-namespace.slang:20]: This test does define the entrypoint inside a struct and a namespace.

Comment (csyonghe) [reply tests/language-feature/namespaces/entrypoint-in-namespace.slang:20]: if things can work for function defined in a struct and a namespace, then it should also work if it is in a struct but not in a namespace, so I don't think we need a separate test for that.

Comment (ArielG-NV) [reply tests/language-feature/namespaces/entrypoint-in-namespace.slang:20]:
```
// My File

struct MyType
{
    static float4 Main(PSInput input) : SV_TARGET
    {
        return input.color;
    }    
}
```

**Pull request merged**